v-next/hardhat/src/internal/builtin-plugins/network-manager/type-validation.ts

```diff
@@ -404,8 +404,8 @@ const userConfigSchema = z.object({
 
 const networkConfigOverrideSchema = z
   .discriminatedUnion("type", [
-    httpNetworkUserConfigSchema.strict(),
-    edrNetworkUserConfigSchema.strict(),
```

Comment:
Change the rules for validating the network config. We are going to allow arbitrary properties to be added to the network config as we intend to do that from plugins (i.e. `hardhat-ledger`).